lib/install/management/finder.go

```diff
@@ -152,6 +155,44 @@ func (d *Dispatcher) GetVCHConfig(vm *vm.VirtualMachine) (*config.VirtualContain
 		return nil, err
 	}
 
+	if vchConfig.IsCreating() {
+		vmRef := vm.Reference()
+		vchConfig.SetMoref(&vmRef)
+	}
+	return vchConfig, nil
+}
+
+func (d *Dispatcher) FetchAndMigrateVCHConfig(vm *vm.VirtualMachine) (*config.VirtualContainerHostConfigSpec, error) {
+	defer trace.End(trace.Begin(""))
+
+	//this is the appliance vm
+	mapConfig, err := vm.FetchExtraConfigBaseOptions(d.ctx)
+	if err != nil {
+		err = errors.Errorf("Failed to get VM extra config of %q: %s", vm.Reference(), err)
+		log.Error(err)
+		return nil, err
+	}
+
+	kv := vmomi.OptionValueMap(mapConfig)
+	newMap, migrated, err := migration.MigrateApplianceConfig(d.ctx, d.session, kv)
+	if err != nil {
+		err = errors.Errorf("Failed to migrate config of %q: %s", vm.Reference(), err)
+		log.Error(err)
+		return nil, err
+	}
+	if !migrated {
+		log.Debugf("No need to migrate configuration for %q", vm.Reference())
+	}
+
+	data := extraconfig.MapSource(newMap)
+	vchConfig := &config.VirtualContainerHostConfigSpec{}
+	result := extraconfig.Decode(data, vchConfig)
+	if result == nil {
+		err = errors.Errorf("Failed to decode migrated VM configuration %q: %s", vm.Reference(), err)
+		log.Error(err)
+		return nil, err
+	}
+
 	//	vchConfig.ID
```

Comment:
Unneeded comment? I realize this wasn't added as part of your changes but we should remove it anyway.